packages/@react-aria/table/docs/useTable.mdx

```diff
@@ -0,0 +1,720 @@
+`@react-stately/table` implements a JSX based interface for building collections instead.
+See [Collection Components](/react-stately/collections.html) for more information,
+and [Collection Interface](/react-stately/Collection.html) for internal details.
+
+In addition, <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableState} />
+manages the state necessary for multiple selection and exposes
+a <TypeLink links={selectionDocs.links} type={selectionDocs.exports.SelectionManager} />,
+which makes use of the collection to provide an interface to update the selection state.
+For more information, see [Selection](/react-stately/selection.html).
+
+## Example
+
+This example uses HTML `<div>` elements to construct the table. It uses `useTableState` to construct the table's collection of rows and columns,
+applying the appropriate attributes to each `<div>` using the various table hooks listed above paired with the div's role in the table.
+The table follows the [Collection Components API](../react-stately/collections.html), accepting both static and dynamic collections.
+
+The example below shows a static collection, which can be used when the full set of table contents is known ahead of time. Be sure to note the usage of
+`Cell`, `Column`, `Row`, `TableBody`, and `TableHeader` to construct the table structure. These elements are similar to `Item` and `Section` used in other components
+following the [Collection Components API](../react-stately/collections.html) mentioned previously, serving as node generators for the table collection.
+
+```tsx example export=true
+import {Cell, Column, Row, TableBody, TableHeader, useTableState} from '@react-stately/table';
+import {mergeProps} from '@react-aria/utils';
+import {useCheckbox} from '@react-aria/checkbox';
+import {useContext, useRef} from 'react';
+import {useFocusRing} from '@react-aria/focus';
+import {useToggleState} from '@react-stately/toggle';
+
+const TableContext = React.createContext(null);
+function useTableContext() {
+  return useContext(TableContext);
+}
+
+function Table(props) {
+  let state = useTableState({...props, showSelectionCheckboxes: true});
+  let ref = useRef();
+
+  let {gridProps} = useTable({
+    ...props,
+    ref: ref
+  }, state);
+
+  let headerRows = state.collection.rows.filter(item => item.type === 'headerrow' && item);
+  let rows = state.collection.rows.filter(item => item.type === 'item' && item);
+
+  let renderColumnNode = (node) => {
+    if (node.props.isSelectionCell) {
+      return <TableSelectAllCell column={node} />;
+    }
+
+    return <TableColumnHeader column={node} />;
+  };
+
+  let renderRowNode = (node) => {
+    if (node.props.isSelectionCell) {
+      return <TableCheckboxCell cell={node} />;
+    }
+
+    if (state.collection.rowHeaderColumnKeys.has(node.column.key)) {
+      return <TableRowHeader cell={node} />;
+    }
+
+    return <TableCell cell={node} />;
+  };
+
```

Comment:
Note that the current aria table example won't bring the focused element into view. I can rectify this by adopting something similar to what `useSelectableList` does with a `useEffect` and `scrollIntoView` but I was wondering if `useGrid` should have that logic inside it if `isVirtualized` is false